contracts/hardhat.config.js

```diff
@@ -195,7 +236,98 @@ module.exports = {
       "MockForwarder",
       "IBosonConfigHandler",
       "IBosonPriceDiscoveryHandler",
-      "Seaport"
+      "Seaport",
+      "OpenSeaWrapper",
+      "OpenSeaWrapperFactory"
     ]
   }
 };
+
+function getWrapperConfig(chainId) {
+  switch (chainId) {
+    case 1: {
+      // Ethereum mainnet
+      if (process.env.ENV_NAME !== "production") {
+        throw new Error(
+          `Environment variable ENV_NAME should be 'production' (current value '${process.env.ENV_NAME}')`
+        );
+      }
+      return {
+        protocolAddress: "0x59A4C19b55193D5a2EAD0065c54af4d516E18Cb5",
+        wethAddress: "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
+        priceDiscoveryClient: "0xb60cf39Fb18e5111174f346d0f39521ef6531fD4",
+        seaport: "0x0000000000000068F116a894984e2DB1123eB395"
+      };
+    }
+    case 137: {
+      // Polygon
+      if (process.env.ENV_NAME !== "production") {
+        throw new Error(
+          `Environment variable ENV_NAME should be 'production' (current value '${process.env.ENV_NAME}')`
+        );
+      }
+      return {
+        protocolAddress: "0x59A4C19b55193D5a2EAD0065c54af4d516E18Cb5",
+        wethAddress: "0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619",
```

Comment:
I think OpenSea only accepts offers in WETH on Polygon, not WMATIC (I know I found it a bit strange, but that's my observation, unless I'm wrong).
Here is the code in opensea-js: https://github.com/ProjectOpenSea/opensea-js/blob/main/src/utils/utils.ts#L176

Reply:
Yes, but this has nothing to do with the `wethAddress` that is passed into the wrapper constructor.

- `wethAddress` in the wrapper is to handle offers that are listed with a native token in Boson. When unwrap happens at the end, Boson protocol can convert them from wrappedNative to native, so the funds in the protocol are always kept in the exchange token that is set on the offer level. If the `wethAddress` is wrong, this will fail:
https://github.com/bosonprotocol/boson-protocol-contracts/blob/b131a972a439ff2d0947db92e86c58703787ef18/contracts/protocol/facets/PriceDiscoveryHandlerFacet.sol#L138-L149 
- given that OS currently does not support the WMATIC offers, it just means that if someone lists a offer with matic on polygon, they won't be able to use it with price discovery
- as for the offers in WETH on polygon, it does not matter, what the value of `wethAddress` in the wrapper is. When an offer with WETH is listed on polygon, that behaves exactly the same as other ERC20 would behave (if they actually supported anything else).

Maybe it looks like this is a bit overengineered, because OS really limits the possibilities. 
But the idea was taken from Zora and Sudoswap wrappers that we made, because those weren't that limiting.

Reply:
I see there is a confusion why we need wethAddress.

In the OpenSeaWrapper contract, it's for when the offer exchangeToken is native (0x00..00) https://github.com/bosonprotocol/core-components/blob/opensea-with-vouchers/contracts/other-contracts/opensea-wrapper/OpenSeaWrapper.sol#L552
However, this usecase can't really happen because OS requires use of WETH
So I think we can remove this line and the wethAddress property from the OpenSeaWrapper contract

While in the rest of the PR, the wethAddress is set in the core-sdk configuration in order to check that any offer used for price-discovery is using WETH (to match the OpenSea constraint). This check is not mandatory, but only here to avoid a wrong usage of the script. However this could be removed (in particular here: https://github.com/bosonprotocol/core-components/blob/a76da7ec47e8334ad2eb2267ef21d44dd77d64d5/scripts/create-price-discovery-offer.ts#L46 because it denies using the script to create a PD offer other than with WETH)